backend/src/apiserver/storage/task_store.go

```diff
@@ -80,6 +97,32 @@ func (s *TaskStore) CreateTask(task *model.Task) (*model.Task, error) {
 	}
 	newTask.UUID = id.String()
 
+	if newTask.CreatedTimestamp == 0 {
+		if newTask.StartedTimestamp == 0 {
```

Comment:
Asked the question before, and I'm still very confused, are there other places where we set create/started time? I recall the answer was yes? But then is it right that we have multiple sources to set created/started time? In what scenarios would we use one source than another?

Reply:
Let's clarify:
`CreatedTimestamp` - time of creating a task entry in DB
`StartedTimestamp` - time of starting an execultion (pod?)
`FinishedTimestamp` - end of execution
`FinishedTimestamp` - `StartedTimestamp` = execution duration (compute time)
`StartedTimestamp` - `CreatedTimestamp` = queue time

The constraint we have:
`StartedTimestamp` <= `StartedTimestamp` <= `FinishedTimestamp`

Tasks can be created when we report workflows or via `TaskService` (reports a individual task). Argo's workflows don't have task-level `CreatedTimestamp`, but have `StartedTimestamp`. `TaskService` may create new tasks without timestamps. This logic take care of this situation and makes sure that we obey the constraint. 

These timestamps can be updated by Persistance Agent later via `CreateOrUpdateTasks` call.